src/pages/Order/index.tsx

```diff
@@ -0,0 +1,123 @@
+import {
+  Box,
+  Button,
+  ChakraProvider,
+  Checkbox,
+  Flex,
+  HStack,
+  Image,
+  Input,
+  Select,
+  Text,
+  Textarea,
+  VStack,
+} from '@chakra-ui/react';
+import type { FieldError } from 'react-hook-form';
+import { useForm } from 'react-hook-form'; // react-hook-form import 추가
+import { useLocation } from 'react-router-dom';
+
+export const Order = () => {
+  const location = useLocation();
+  const { productDetail, quantity, price } = location.state || {};
+
+  const {
+    register,
+    handleSubmit,
+    watch,
+    formState: { errors },
+  } = useForm(); // useForm 훅을 사용하여 폼 상태와 메소드 초기화
+
+  // ref 대신 register 함수 사용
+  const messageRef = register('message', {
+    required: '카드 메시지를 입력해주세요.',
+    maxLength: { value: 100, message: '카드 메시지는 100자 이내로 입력해주세요.' },
+  });
+  const receiptNumberRef = register('receiptNumber', {
+    required: '현금영수증 번호를 입력해주세요.',
+    pattern: { value: /^\d+$/, message: '현금영수증 번호는 숫자만 입력해주세요.' },
+  });
+
+  const onSubmit = (data: Record<string, null>) => {
```

Comment:
```suggestion
  const onSubmit: SubmitHandler<FormValues> = (data) => {
```

[handleSubmit](https://react-hook-form.com/docs/useform/handlesubmit) 공식문서에 보면 submit handler의 타입을 따로 제공해주고 있으므로 이를 활용하는게 좋겠습니다.